pkg/scheduler/factory/factory.go

```diff
@@ -681,7 +682,8 @@ func MakeDefaultErrorFunc(client clientset.Interface, podQueue internalqueue.Sch
 				pod, err := client.CoreV1().Pods(podID.Namespace).Get(podID.Name, metav1.GetOptions{})
```

Comment:
this infinite loop is hard to read and reason about, I am pretty sure there is a waitUntil function that we can use instead of this logic, would be great if we can clean this up in a followup PR.

Reply:
SG. There is a comment for this `Note that this is extremely rudimentary and we need a more real error handling path.` There might be a chance to further refactor this.

Reply:
I am going to create a PR for this.